src/appleseed/renderer/kernel/lighting/pathtracer.h

```diff
@@ -768,30 +768,33 @@ bool PathTracer<PathVisitor, VolumeVisitor, Adjoint>::march(
         }
 
         // Retrieve extinction spectrum.
-        Spectrum extinction_coef = volume->extinction_coefficient(
-            vertex.m_volume_data, volume_ray);
-        const size_t channel_count = std::max(extinction_coef.size(), vertex.m_throughput.size());
-        if (extinction_coef.size() != channel_count)
-            Spectrum::upgrade(extinction_coef, extinction_coef);
+        const Spectrum& extinction_coef =
+            volume->extinction_coefficient(vertex.m_volume_data, volume_ray);
         
         sampling_context.split_in_place(1, 2);
 
         // Sample channel uniformly at random.
         const float s = sampling_context.next2<float>();
-        const size_t channel = foundation::truncate<size_t>(s * channel_count);
-        const bool extinction_is_null = (extinction_coef[channel] < 1.0e-6f);
+        const size_t channel = foundation::truncate<size_t>(s * Spectrum::size());
+        const bool extinction_is_null = extinction_coef[channel] < 1.0e-6f;
 
-        float distance_sample = 0.0f;
-        float distance_pdf = 0.0f;
-        if (!extinction_is_null)
+        // Sample distance.
```

Comment:
I suggest to remove the comment or improve it. The original comment position looks clearer IMHO.